File: patch_gallery.py
import codecs

with codecs.open(r'c:\Users\hp\Jsroom-\src\Screens\GalleryScreen.js', 'r', 'utf-8') as f:
    lines = f.readlines()

with codecs.open(r'c:\Users\hp\Jsroom-\gallery_code2.txt', 'r', 'utf-8') as f:
    replacement = f.read()

react_icons_import = """import { 
  FaImages, 
  FaExpand,
  FaTimes,
  FaChevronLeft,
  FaChevronRight,
  FaCalendarAlt,
  FaArrowRight,
  FaFilter,
  FaSearch,
  FaPhone,
  FaEnvelope,
  FaMapMarkerAlt,
  FaInstagram,
  FaTwitter,
  FaFacebookF,
  FaPlay
} from 'react-icons/fa';
"""

gallery_idx = -1
for i, line in enumerate(lines):
    if line.startswith('const galleryImages = ['):
        gallery_idx = i
        break

end_idx = -1
if gallery_idx != -1:
    for i in range(gallery_idx, len(lines)):
        if lines[i].startswith('];'):
            end_idx = i
            break

if gallery_idx != -1 and end_idx != -1:
    new_lines = lines[:3] # Keep up to line 3 (index 2)
    new_lines.append('\n' + react_icons_import + '\n')
    new_lines.append('// Gallery images\n')
    new_lines.append(replacement)
    new_lines.append('\n')
    new_lines.extend(lines[end_idx+1:])
    
    with codecs.open(r'c:\Users\hp\Jsroom-\src\Screens\GalleryScreen.js', 'w', 'utf-8') as f:
        f.writelines(new_lines)
    print("Replaced content!")
else:
    print("Could not find boundaries", gallery_idx, end_idx)

File: patch_gallery.js
const fs = require('fs');

const lines = fs.readFileSync('c:\\Users\\hp\\Jsroom-\\src\\Screens\\GalleryScreen.js', 'utf-8').split('\n');
const replacement = fs.readFileSync('c:\\Users\\hp\\Jsroom-\\gallery_code2.txt', 'utf-8');

const reactIconsImport = `import { 
  FaImages, 
  FaExpand,
  FaTimes,
  FaChevronLeft,
  FaChevronRight,
  FaCalendarAlt,
  FaArrowRight,
  FaFilter,
  FaSearch,
  FaPhone,
  FaEnvelope,
  FaMapMarkerAlt,
  FaInstagram,
  FaTwitter,
  FaFacebookF,
  FaPlay
} from 'react-icons/fa';`;

let galleryIdx = -1;
for (let i = 0; i < lines.length; i++) {
  if (lines[i].startsWith('const galleryImages = [')) {
    galleryIdx = i;
    break;
  }
}

let endIdx = -1;
if (galleryIdx !== -1) {
  for (let i = galleryIdx; i < lines.length; i++) {
    if (lines[i].startsWith('];')) {
      endIdx = i;
      break;
    }
  }
}

if (galleryIdx !== -1 && endIdx !== -1) {
  const newLines = lines.slice(0, 3);
  newLines.push('\n' + reactIconsImport + '\n');
  newLines.push('// Gallery images');
  newLines.push(replacement);
  newLines.push('\n');
  newLines.push(...lines.slice(endIdx + 1));
  
  fs.writeFileSync('c:\\Users\\hp\\Jsroom-\\src\\Screens\\GalleryScreen.js', newLines.join('\n'));
  console.log("Replaced content!");
} else {
  console.log("Could not find boundaries", galleryIdx, endIdx);
}
